### ISSUE TITLE
Gaia Sync Panic

### ISSUE THREAD

**Issue by jungyeon0904** [cosmos/gaia#988]
## Summary of Bug

When I start cosmos synchronization, panic occurs and ends :
<pre><code>panic: failed to write batch: leveldb/table: corruption on data-block (pos=31076): checksum mismatch, want=0x746fc31c got=0x71a75600 [file=1459199.ldb]</code></pre>

More details are as follows:
<pre><code>
root@comnet-System-Product-Name:~/gaia# gaiad start --p2p.seeds bf8328b66dceb4987e5cd94430af66045e59899f@public-seed.cosmos.vitwit.com:26656,cfd785a4224c7940e9a10f6c1ab24c343e923bec@164.68.107.188:26656,d72b3011ed46d783e369fdf8ae2055b99a1e5074@173.249.50.25:26656,ba3bacc714817218562f743178228f23678b2873@public-seed-node.cosmoshub.certus.one:26656,3c7cad4154967a294b3ba1cc752e40e8779640ad@84.201.128.115:26656,366ac852255c3ac8de17e11ae9ec814b8c68bddb@51.15.94.196:26656 --x-crisis-skip-assert-invariants
3:19PM INF starting ABCI with Tendermint
3:19PM INF Starting multiAppConn service impl=multiAppConn module=proxy
3:19PM INF Starting localClient service connection=query impl=localClient module=abci-client
3:19PM INF Starting localClient service connection=snapshot impl=localClient module=abci-client
3:19PM INF Starting localClient service connection=mempool impl=localClient module=abci-client
3:19PM INF Starting localClient service connection=consensus impl=localClient module=abci-client
3:19PM INF Starting EventBus service impl=EventBus module=events
3:19PM INF Starting PubSub service impl=PubSub module=pubsub
3:19PM INF Starting IndexerService service impl=IndexerService module=txindex
3:19PM INF ABCI Handshake App Info hash="�?�����:ծ�m=R��(]�\f44o\"S��1�\x00��" height=5565149 module=consensus protocol-version=0 software-version=
3:19PM INF ABCI Replay Blocks appHeight=5565149 module=consensus stateHeight=5565149 storeHeight=5565150
3:19PM INF Replay last block using real app module=consensus
3:19PM INF minted coins from module account amount=4333438uatom from=mint module=x/bank
3:19PM INF executed block height=5565150 module=consensus num_invalid_txs=0 num_valid_txs=0
panic: failed to write batch: leveldb/table: corruption on data-block (pos=31076): checksum mismatch, want=0x746fc31c got=0x71a75600 [file=1459199.ldb]

goroutine 1 [running]:
github.com/cosmos/cosmos-sdk/store/iavl.(*Store).Commit(0xc0001e83d0)
        github.com/cosmos/cosmos-sdk@v0.42.4/store/iavl/store.go:113 +0x165
github.com/cosmos/cosmos-sdk/store/rootmulti.commitStores(0x54eade, 0xc000f16e40)
        github.com/cosmos/cosmos-sdk@v0.42.4/store/rootmulti/store.go:919 +0x150
github.com/cosmos/cosmos-sdk/store/rootmulti.(*Store).Commit(0xc0001f0280)
        github.com/cosmos/cosmos-sdk@v0.42.4/store/rootmulti/store.go:344 +0x6f
github.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).Commit(0xc001703380)
        github.com/cosmos/cosmos-sdk@v0.42.4/baseapp/abci.go:293 +0x229
github.com/tendermint/tendermint/abci/client.(*localClient).CommitSync(0xc001030c00)
        github.com/tendermint/tendermint@v0.34.9/abci/client/local_client.go:258 +0xb6
github.com/tendermint/tendermint/proxy.(*appConnConsensus).CommitSync(0x2bee170)
        github.com/tendermint/tendermint@v0.34.9/proxy/app_conn.go:93 +0x22
github.com/tendermint/tendermint/state.(*BlockExecutor).Commit(_, {{{0xb, 0x0}, {0xc001a85960, 0x7}}, {0xc001a85970, 0xb}, 0x4f5b97, 0x54eade, {{0xc000da4140, ...}, ...}, ...}, ...)
        github.com/tendermint/tendermint@v0.34.9/state/execution.go:228 +0x269
github.com/tendermint/tendermint/state.(*BlockExecutor).ApplyBlock(_, {{{0xb, 0x0}, {0xc001a85960, 0x7}}, {0xc001a85970, 0xb}, 0x4f5b97, 0x54eade, {{0xc000da4140, ...}, ...}, ...}, ...)
        github.com/tendermint/tendermint@v0.34.9/state/execution.go:180 +0x6ee
github.com/tendermint/tendermint/consensus.(*Handshaker).replayBlock(_, {{{0xb, 0x0}, {0xc001a85960, 0x7}}, {0xc001a85970, 0xb}, 0x4f5b97, 0x54eadd, {{0xc0016b6220, ...}, ...}, ...}, ...)
        github.com/tendermint/tendermint@v0.34.9/consensus/replay.go:503 +0x23c
github.com/tendermint/tendermint/consensus.(*Handshaker).ReplayBlocks(_, {{{0xb, 0x0}, {0xc001a85960, 0x7}}, {0xc001a85970, 0xb}, 0x4f5b97, 0x54eadd, {{0xc0016b6220, ...}, ...}, ...}, ...)
        github.com/tendermint/tendermint@v0.34.9/consensus/replay.go:416 +0x7ae
github.com/tendermint/tendermint/consensus.(*Handshaker).Handshake(0xc0019a4510, {0x1cdb000, 0xc00013e8f0})
        github.com/tendermint/tendermint@v0.34.9/consensus/replay.go:268 +0x3c8
github.com/tendermint/tendermint/node.doHandshake({_, _}, {{{0xb, 0x0}, {0xc001a85960, 0x7}}, {0xc001a85970, 0xb}, 0x4f5b97, 0x54eadd, ...}, ...)
        github.com/tendermint/tendermint@v0.34.9/node/node.go:293 +0x1b8
github.com/tendermint/tendermint/node.NewNode(0xc0010a6500, {0x1c94130, 0xc0001f12c0}, 0xc0001e8ce0, {0x1c79340, 0xc0025d8db0}, 0x412854, 0xc001144118, 0xc0001e8f50, {0x1cb31b8, ...}, ...)
        github.com/tendermint/tendermint@v0.34.9/node/node.go:700 +0x578
github.com/cosmos/cosmos-sdk/server.startInProcess(_, {{0x0, 0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x1cba650, 0xc0010900e0}, ...}, ...)
        github.com/cosmos/cosmos-sdk@v0.42.4/server/start.go:244 +0x625
github.com/cosmos/cosmos-sdk/server.StartCmd.func2(0xc0010b4000, {0xc0010f9fb0, 0x0, 0x3})
        github.com/cosmos/cosmos-sdk@v0.42.4/server/start.go:120 +0x168
github.com/spf13/cobra.(*Command).execute(0xc0010b4000, {0xc0010f9f50, 0x3, 0x3})
        github.com/spf13/cobra@v1.1.3/command.go:852 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc00104af00)
        github.com/spf13/cobra@v1.1.3/command.go:960 +0x3ad
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.1.3/command.go:897
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        github.com/spf13/cobra@v1.1.3/command.go:890
github.com/cosmos/cosmos-sdk/server/cmd.Execute(0x0, {0xc001047aa0, 0xb})
        github.com/cosmos/cosmos-sdk@v0.42.4/server/cmd/execute.go:36 +0x1d0
main.main()
        github.com/cosmos/gaia/v4/cmd/gaiad/main.go:16 +0x2c

</code></pre>
## Version

root@comnet-System-Product-Name:~/gaia# gaiad version --long
name: gaia
server_name: gaiad
version: v4.2.1
commit: dbd8a6fb522c571debf958837f9113c56d418f6b
build_tags: netgo,ledger
go: go version go1.17.1 linux/amd64

## Steps to Reproduce

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by yaruwangway** [comment]
can you tell me more info. ? Do you have some block data already before running gaia v4.2.1 ?
if you want to sync to the latest block, you might need v5.0.5.

**Comment by jungyeon0904** [comment]
I installed Gaia according to the instructions in the following link.
https://github.com/cosmos/gaia/blob/main/docs/gaia-tutorials/installation.md

And I wanted to participate in the mainnet hub, so I was synchronizing up to 6910000 height blocks.
So I installed Gaia 4.2.1 version, and the instructions used the following link.
https://github.com/cosmos/mainnet/blob/master/README.md

During synchronization, there was a precursor phenomenon that could not be connected to several peers.
I had re-entered the $gaiad start command.

The following panic occurred after synchronization retardation occurred as several peers were disconnected and reconnected.
~~~
panic: failed to write batch: leveldb/table: corruption on data-block (pos=31076): checksum mismatch, want=0x746fc31c got=0x71a75600 [file=1459199.ldb]
~~~
As explained in Panic, it seems that there was a twist in creating DB during synchronization.

To solve this problem, the problematic file 1459199.ldb was deleted and the schema was held again.

But my solution was not enough. There is a panic that can't find DB at all.

Now, all of them are initialized, and Gaia has been downloaded from the beginning and synchronized.

However, I am worried that this problem will occur again.
